### PR TITLE
add infrastructure for watch tests

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
test infrastructure

**Did you add tests for your changes?**
yes

**If relevant, link to documentation update:**
n/a

**Summary**
adds infrastructure for watch tests and some basic tests

**Does this PR introduce a breaking change?**
no

### CODE REVIEW

File: test/WatchTestCases.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+describe("WatchTestCases", function() {
+	var casesPath = path.join(__dirname, "watchCases");
+	var categories = fs.readdirSync(casesPath);
+
+	categories = categories.map(function(cat) {
+		return {
+			name: cat,
+			tests: fs.readdirSync(path.join(casesPath, cat)).filter(function(folder) {
+				return folder.indexOf("_") < 0;
```
Comment (TheLarkInn) [inline test/WatchTestCases.test.js]: This for .DS_Store? Maybe it's safer to filter on fsStatSync.isDirectory().

File: test/WatchTestCases.test.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+										}).join("\n");
+									} else {
+										var p = path.join(currentDirectory, module);
+										content = fs.readFileSync(p, "utf-8");
+									}
+									fn = vm.runInThisContext("(function(require, module, exports, __dirname, __filename, it, WATCH_STEP) {" + content + "\n})", p);
+									var module = {
+										exports: {}
+									};
+									fn.call(module.exports, _require.bind(null, path.dirname(p)), module, module.exports, path.dirname(p), p, _it, run.name);
```
Comment (TheLarkInn) [inline test/WatchTestCases.test.js]: Is this potentially why ci is failing trying to access `should`

Comment (sokra) [inline test/WatchTestCases.test.js]: Could you elaborate?

Comment (TheLarkInn) [inline test/WatchTestCases.test.js]: Maybe its line 141 thats the one in question. But do you have to pass the required `should` in as a variable in the `vm.runInThisContext()`, would make sense why `it` works but not `should`. This is just a hunch, tbh.

**Pull request merged**